tex/chProgramming.tex

```diff
@@ -1068,19 +1071,17 @@ \subsection{Recursion on natural numbers}\label{ssec:recnat}
 \(n\) to \(\max\left\{m-n, 0\right\}\) as opposed to \(m-n\).
 
 
- Thus, mathematically speaking,
-\C{subn} sends two naturals \(m\) and \(n\) to
-\(\max\left\{m-n, 0\right\}\) as opposed to \(m-n\).
-This oddity is imposed by
-the fact that any function defined in \Coq{} has to be total, i.e., has
-to assign a value to any element in the type of its arguments. Since
-\C{subn} has type \C{nat -> nat -> nat},
-an element \emph{in type} \C{nat} has to be
-output by the function even in the case when  the second argument is
-non-zero and the first argument is zero. This problem disappears with
+This problem disappears with
 the introduction of another type of numbers, with negative integers.
 (Alternatively, it is possible to implement partial functions by
 extending the output type; see Section~\ref{sec:othercontainers}.)
+Actually, the strange behavior of the \C{subn} function described
+above has its virtues:  In particular, it allows defining the
+less-than-or-equal relation \C{leq} on naturals as follows:
+\begin{coq}{name=nat_leq_def}{}
+Definition leq m n := m - n == 0.
+\end{coq}
+And this is indeed how \C{leq} is defined in \mcbMC{}.
```

Comment:
All changes to tex/chProgramming.tex, but for this one, have been integrated in the branch "modifs-on-chapter1" that constitutes PR #31